tox.ini

```diff
@@ -3,123 +3,40 @@ minversion = 3.5.0
-deps =
-    -r{toxinidir}/server/requirements.txt
-    -r{toxinidir}/server/test-requirements.txt
-commands =
-    bash -c "./server/bin/unittests {posargs}"
-
-[testenv:datalog]
-description = Runs all non-Python3-based agent tool-scripts/datalog unit/functional tests
-deps =
-    -r{toxinidir}/agent/requirements.txt
-    -r{toxinidir}/agent/test-requirements.txt
-commands =
-    bash -c "./agent/tool-scripts/datalog/unittests {posargs}"
-
-[testenv:postprocess]
-description = Runs all non-Python3-based agent tool-scripts/postprocess unit/functional tests
-deps =
-    -r{toxinidir}/agent/test-requirements.txt
-commands =
-    bash -c "./agent/tool-scripts/postprocess/unittests {posargs}"
-
-[testenv:tool-scripts]
-description = Runs all non-Python3-based agent tool-scripts unit/functional tests
-deps =
-    -r{toxinidir}/agent/test-requirements.txt
-commands =
-    bash -c "./agent/tool-scripts/unittests {posargs}"
-
-[testenv:util-scripts]
-description = Runs all non-Python3-based agent util-scripts unit/functional tests
 deps =
+    -r{toxinidir}/lint-requirements.txt
     -r{toxinidir}/agent/requirements.txt
     -r{toxinidir}/agent/test-requirements.txt
-commands =
-    bash -c "./agent/util-scripts/unittests {posargs}"
-
-[testenv:bench-scripts]
-description = Runs all non-Python3-based agent bench-scripts unit/functional tests
-deps =
-    -r{toxinidir}/agent/requirements.txt
-    -r{toxinidir}/agent/test-requirements.txt
-commands =
-    bash -c "./agent/bench-scripts/unittests {posargs}"
-
-[testenv:py3-server]
-description = Runs all Python3-based server unit and functional tests
-deps =
     -r{toxinidir}/server/requirements.txt
     -r{toxinidir}/server/test-requirements.txt
-commands =
-    pytest --cov --cov-append \
-      --cov-report html:{envdir}/cov/html {posargs} \
-      ./lib/pbench/test/unit/common \
-      ./lib/pbench/test/unit/server
-
-[testenv:py3-agent]
-description = Runs all Python3-based agent unit and functional tests
-deps =
-    -r{toxinidir}/agent/requirements.txt
-    -r{toxinidir}/agent/test-requirements.txt
-commands =
-    pytest --cov --cov-append \
-      --cov-report html:{envdir}/cov/html {posargs} \
-      ./lib/pbench/test/unit/common \
-      ./lib/pbench/test/unit/agent \
-      ./lib/pbench/test/functional
-
-[testenv:lint]
-description = Runs all linting tasks
+commands_pre =
+    {toxinidir}/utils/detox {envdir}
 commands =
     black --check .
     flake8 .
-deps = -r{toxinidir}/lint-requirements.txt
-skip_install = true
-usedevelop = false
-
-[testenv:jenkins-unittests]
-description = Runs all external tests from one tox environment
-commands =
-    bash -c "./jenkins/run-unittests {envdir}"
-
-[testenv:jenkins-pytests]
-description = Runs all external tests from one tox environment
-commands =
-    bash -c "./jenkins/run-pytests {envdir}"
+    bash -c "{toxinidir}/exec-unittests {envdir} {posargs}"
+whitelist_externals =
+    bash
+    rm
+    find
+    env
```

Comment:
By the way, apparently `whitelist_externals` is [now deprecated](https://tox.wiki/en/latest/config.html#conf-allowlist_externals) -- it's been respelled `allowlist_externals`.

Apparently this is supposed to be a list of commands that we use in the `commands` section that we don't want to be warned about...I'm thinking that `rm`, `find`, and `env` aren't in that set anymore (and, I'm wondering if `detox` doesn't need to be added)....

Reply:
Not sure what is going on, but when I use `allowlist_externals`, I can the traceback as see in http://jenkins.intlab.perf-infra.lab.eng.rdu2.redhat.com/job/pbench-ci-multi/job/PR-2551/41/console.